### PR TITLE
[PageActions] Add OutlineAction for ComplexActions

### PR DESCRIPTION
**Pull request by ashwinnarayanan2001**
### WHY are these changes introduced?
This change was introduced because destructive buttons located in the PageActions component in admin require an outline prop (new design language). This is mentioned in #Shopify/polaris-ux#479. 

When we are adding the outline prop to the PageActions component, the PageActions component imports a type called ComplexAction, which requires an OutlineableAction to be defined. 

### WHAT is this pull request doing?
This pull request adds an OutlineableAction that is added onto the ComplexAction type (which is being used by the PageActions component). 

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}

```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -159,6 +159,11 @@ export interface LoadableAction extends Action {
   loading?: boolean;
 }
 
+export interface OutlineableAction extends Action {
+  /** Should a outline be displayed */
```
Comment (kyledurand) [inline src/types.ts:163]:
```suggestion
  /** Should action be displayed as an outlined button */
```

**Review by alex-page — APPROVED** [review]
Nice, thanks Ashwin!

**Review by kyledurand — COMMENTED** [review]
Before shipping can you build this for web and test that TS is happy with this? Instructions [here](https://github.com/Shopify/polaris-react#testing-in-a-consuming-project)

**Comment by kyledurand** [comment]
Make sure to add to the changelog for this as well

File: UNRELEASED.md [hunk 2]
```diff
@@ -8,6 +8,9 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 
 ### Enhancements
 
+- Added `RefOject` as a possible type for the `activator` prop on `Modal` ([#3395](https://github.com/Shopify/polaris-react/pull/3395))
```
Comment (kyledurand) [inline UNRELEASED.md:11]: You can remove this line, it was already released

**Review by kyledurand — APPROVED** [review]
Nice! Just make sure to remove the accidental addition to unreleased

**Pull request merged**